core/document/src/color.rs

```diff
@@ -16,21 +21,51 @@ impl Color {
 	pub const GREEN: Color = Color::from_unsafe(0., 1., 0.);
 	pub const BLUE: Color = Color::from_unsafe(0., 0., 1.);
 
+	/// return Some(Color) if `red`, `green`, `blue` and `alpha` have a valid value. Negative number (including -0.0), Nan and infinity are not valid value and return None.
```

Comment:
"Return" -> "Returns". Put `Some(Color)` in backticks. "Negative number" -> "Negative numbers". Put `-0.0` in backticks. Should "Nan" be "`f32::NAN`"? Otherwise "Nan" -> "NaN". "not valid value" -> "not valid values". Backtick "None".